### DISCUSSION TITLE
Intellij like "Compare with Git Branch/Revision and local history" feature

### DISCUSSION THREAD

**Discussion by shivamka1** [zed-industries/zed#23793] [Feature Requests]
### Description

It will be really nice if we can compare files we are working on with revision/branches on github without having to leave the editor. It is really handy. I love this feature in Intellij and miss it in editors like VSCode. Intellij also allows to compare with local history it maintains of every file.

I would love to see these features in Zed.

### If applicable, add mockups / screenshots to help present your vision of the feature

**Compare with Git Revision/Branch**
<img width="566" alt="image" src="https://github.com/zed-industries/community/assets/4599890/334e624d-8a75-4513-b5f0-1f74e28bcec7">

**Compare with Local History**
<img width="526" alt="image" src="https://github.com/zed-industries/community/assets/4599890/81c4f05d-1792-4100-9841-61b877477452">

**Comment by egor-a-trubnikov-panov** [comment]
Additionally, it would be very useful to have a “Compare with Clipboard” feature, as well as a simple way to compare any two files directly within the editor.

**Comment by robaho** [comment]
The git history/diff features in Zed are pretty limited. I have been trying it out recently - I use VSCode and XCode - and its a shame.
